website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,223 @@
+
+First, clone the repo:
+
+```sh
+git clone https://github.com/alvinometric/feature-flags-ai
+cd feature-flags-ai
+```
+
+Install the dependencies:
+
+```sh
+npm install
+```
+
+Run the app:
+
+```sh
+npm run dev
+```
+
+You should now see a chatbot UI, like this:
+
+![chatbot UI](./sveltekit-chatbot.png)
+
+This is a simple Sveltekit app with a chatbot interface. If you’re not familiar with SvelteKit, it works similarly to frameworks like Next.js, Nuxt, or SolidStart, where your file structure defines the routing.
+
+The most important file for this tutorial is the `src/routes/api/chat/+server.js` file.
+
+It creates an API endpoint at `/api/chat`. When your frontend sends a POST request to `/api/chat`, this is the code that handles the request.
+
+```javascript
+import { createOpenAI } from "@ai-sdk/openai";
+import { streamText } from "ai";
+import { env } from "$env/dynamic/private";
+
+const openai = createOpenAI({
+    apiKey: env.OPENAI_API_KEY ?? "",
+});
+
+const variants = {
+    variant1: {
+        model: "gpt-3.5-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant2: {
+        model: "gpt-4-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant3: {
+        model: "gpt-4-vision-preview",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+};
+
+export const POST = async ({ request }) => {
+    const { messages } = await request.json();
+    const variant = variants["variant1"];
+    const result = await streamText({
+        model: openai(variant.model),
+        messages,
+        max_tokens: variant.max_tokens,
+        temperature: variant.temperature,
+    });
+    return result.toDataStreamResponse();
+};
+```
+
+This file is doing a few key things:
+
+1. Sets up our OpenAI client with an API key.
+2. Defines different AI model configurations in the `variants` object.
+3. Handles incoming chat requests with the `POST` function.
+
+When a request comes in, it:
+
+-   Extracts the messages from the request body.
+-   Selects a variant (currently hardcoded to 'variant1').
+-   Uses the OpenAI API to generate a response.
+-   Streams the response back to the client.
+
+The `streamText` function is one of the utilities provided by Vercel's AI SDK. It helps deal with the real-time streaming of the AI's responses.
+
+## Creating a feature flag with AI variants
+
+Instead of hardcoding `variant1`, we want to use feature flags to dynamically choose which AI model to use. This will let us easily switch between models, test different configurations, or even do some A/B testing to see which model performs better for which task.
+
+To implement this, we'll need to:
+
+1. Set up a feature flag provider (we'll use Unleash).
+2. Replace our static variant selection with feature flag calls.
```

Comment:
"Calls" might sound a bit too loaded. It’s just a single "call" that returns a variant.